org.eclipse.xtext/src/org/eclipse/xtext/resource/ResourceServiceProviderServiceLoader.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2016 TypeFox GmbH (http://www.typefox.io) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.resource;
+
+import java.util.ServiceLoader;
+
+import org.eclipse.xtext.ISetup;
+import org.eclipse.xtext.resource.IResourceServiceProvider.Registry;
+import org.eclipse.xtext.resource.impl.ResourceServiceProviderRegistryImpl;
+
+import com.google.inject.Injector;
+import com.google.inject.Provider;
+import com.google.inject.Singleton;;
+
+/**
+ * @author Sven Efftinge - Initial contribution and API
+ * @since 2.11
+ */
+@Singleton
+public class ResourceServiceProviderServiceLoader implements Provider<Registry> {
+	
+	private ServiceLoader<ISetup> setupLoader = ServiceLoader.load(ISetup.class);
```

Comment:
Can be final, not sure if it's allowed to be static?

Reply:
Will make it final only.